app/models/product_import/entry_validator.rb

```diff
@@ -220,6 +232,25 @@ def mark_as_existing_variant(entry, existing_variant)
       end
     end
 
+    def product_field_errors(entry, existing_product)
+      non_updatable_fields.each do |display_name, attribute|
+        next if attributes_match?(attribute, existing_product, entry) || attributes_blank?(attribute, existing_product, entry)
+        mark_as_invalid(entry, attribute: display_name, error: I18n.t('admin.product_import.model.not_updatable'))
+      end
+    end
+
+    def non_updatable_fields
+      EntryValidator.non_updatable_fields
+    end
+
+    def attributes_match?(attribute, existing_product, entry)
+      existing_product.send(attribute) == entry.send(attribute)
+    end
+
+    def attributes_blank?(attribute, existing_product, entry)
+      existing_product.send(attribute).blank? && entry.send(attribute).blank?
```

Comment:
please, use `public_send` over `send`. The latter does not respect `private` methods.